catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/AimidbAttribute.java

```diff
@@ -0,0 +1,104 @@
+
+import java.io.Serializable;
+import java.util.function.Function;
+
+import org.codice.imaging.nitf.core.tre.Tre;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
+
+public enum AimidbAttribute implements NitfAttribute<Tre> {
+    ACQUISITION_DATE_AND_TIME("acquisitionDateAndTime",
+            "ACQUISITION_DATE",
+            tre -> TreUtility.getTreValue(tre, "ACQUISITION_DATE")),
+    MISSION_NUMBER("missionNumber", "MISSION_NO", tre -> TreUtility.getTreValue(tre, "MISSION_NO")),
+    MISSION_IDENTIFICATION("missionIdentification",
+            "MISSION_IDENTIFICATION",
+            tre -> TreUtility.getTreValue(tre, "MISSION_IDENTIFICATION")),
+    FLIGHT_NUMBER("flightNumber", "FLIGHT_NO", tre -> TreUtility.getTreValue(tre, "FLIGHT_NO")),
+    IMAGE_OPERATION_NUMBER("imageOperationNumber",
+            "OP_NUM",
+            tre -> TreUtility.getTreValue(tre, "OP_NUM")),
+    CURRENT_SEGMENT_ID("currentSegementId",
+            "CURRENT_SEGMENT",
+            tre -> TreUtility.getTreValue(tre, "CURRENT_SEGMENT")),
+    REPROCESS_NUMBER("reprocessNumber",
+            "REPRO_NUM",
+            tre -> TreUtility.getTreValue(tre, "REPRO_NUM")),
+    REPLAY("replay", "REPLAY", tre -> TreUtility.getTreValue(tre, "REPLAY")),
+    START_TILE_COLUMN_NUMBER("startTileColumnNumber",
+            "START_TILE_COLUMN",
+            tre -> TreUtility.getTreValue(tre, "START_TILE_COLUMN")),
+    START_TILE_ROW_NUMBER("startTileRowNumber",
+            "START_TILE_ROW",
+            tre -> TreUtility.getTreValue(tre, "START_TILE_ROW")),
+    ENDING_SEGMENT("endingSegment",
+            "END_SEGMENT",
+            tre -> TreUtility.getTreValue(tre, "END_SEGMENT")),
+    END_TILE_COLUMN_NUMBER("endTileColumnNumner",
+            "END_TILE_COLUMN",
+            tre -> TreUtility.getTreValue(tre, "END_TILE_COLUMN")),
+    END_TILE_ROW_NUMBER("endTileRowNumber",
+            "END_TILE_COLUMN",
+            tre -> TreUtility.getTreValue(tre, "END_TILE_ROW")),
+    COUNTRY("country", "COUNTRY", tre -> TreUtility.getTreValue(tre, "COUNTRY")),
+    LOCATION("location", "LOCATION", tre -> TreUtility.getTreValue(tre, "LOCATION"));
+
+    private static final String ATTRIBUTE_NAME_PREFIX = "nitf.aimidb.";
+
+    private String shortName;
+
+    private String longName;
+
+    private Function<Tre, Serializable> accessorFunction;
+
+    private AttributeDescriptor attributeDescriptor;
+
+    AimidbAttribute(String longName, String shortName,
+            Function<Tre, Serializable> accessorFunction) {
+        this.longName = longName;
+        this.shortName = shortName;
+        this.accessorFunction = accessorFunction;
+        this.attributeDescriptor = new AttributeDescriptorImpl(ATTRIBUTE_NAME_PREFIX + longName,
+                true,
+                true,
+                false,
+                true,
+                BasicTypes.STRING_TYPE);
```

Comment:
Are all of these attributes be stored as a string type?  This doesn't look right to me.